core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3265,6 +3265,98 @@ protected Scope visitSetOperation(SetOperation node, Optional<Scope> scope)
             return createAndAssignScope(node, scope, outputDescriptorFields);
         }
 
+        @Override
+        protected Scope visitUnion(Union node, Optional<Scope> scope)
```

Comment:
Any specific reason on applying it only for `Union` node ? Why not specific to SetOperation node ?